pyanaconda/configuration.py

```diff
@@ -0,0 +1,222 @@
+
+    @property
+    def ibft(self):
+        """Enable iBFT usage during the installation."""
+        return self._get_option("ibft", bool)
+
+    @property
+    def gpt(self):
+        """Do you prefer creation of GPT disk labels?"""
+        return self._get_option("gpt", bool)
+
+    @property
+    def multipath_friendly_names(self):
+        """Use user friendly names for multipath devices."""
+        return self._get_option("multipath_friendly_names", bool)
+
+
+class AnacondaConfiguration(object):
+    """Representation of the Anaconda configuration."""
+
+    @classmethod
+    def from_defaults(cls):
+        """Get the default Anaconda configuration.
+
+        Read the current configuration from the temporary config file.
+        Or load the configuration from the configuration directory.
+
+        :return: an instance of AnacondaConfiguration
+        """
+        config = cls()
+
+        # Read the temporary configuration file.
+        config_path = os.environ.get("ANACONDA_CONFIG_TMP", ANACONDA_CONFIG_TMP)
+
+        # Or use the defaults if it doesn't exist.
+        if not config_path or not os.path.exists(config_path):
+            config_dir = os.environ.get("ANACONDA_CONFIG_DIR", ANACONDA_CONFIG_DIR)
+            config_path = os.path.join(config_dir, "anaconda.conf")
+
+        config.read(config_path)
+
+        # Validate the configuration.
+        config.validate()
+        return config
+
+    def __init__(self):
+        """Initialize the configuration."""
+        self._sources = []
+        self._parser = create_parser()
+
+        self._anaconda = AnacondaSection("Anaconda", self.get_parser())
+        self._storage = StorageSection("Storage", self.get_parser())
+        self._services = ServicesSection("Services", self.get_parser())
+
+    @property
+    def anaconda(self):
+        """The Anaconda section."""
+        return self._anaconda
+
+    @property
+    def storage(self):
+        """The Storage section."""
+        return self._storage
+
+    @property
+    def services(self):
+        """The Services section."""
+        return self._services
+
+    def get_parser(self):
+        """Get the configuration parser.
+
+        :return: instance of the ConfigParser
+        """
+        return self._parser
+
+    def get_sources(self):
+        """Get the configuration sources.
+
+        :return: a list of file names
+        """
+        return self._sources
+
+    def read(self, path):
+        """Read a configuration file.
+
+        :param path: a path to the file
+        """
+        read_config(self._parser, path)
+        self._sources.append(path)
+
+    def write(self, path):
+        """Write a configuration file.
+
+        :param path: a path to the file
+        """
+        write_config(self._parser, path)
+
+    def validate(self):
+        """Validate the configuration."""
+        self._validate_members(self)
+
+    def _validate_members(self, obj):
```

Comment:
I would remove the `obj` object. It adds superfluous complexity.

Reply:
Let's use obj=self.

Reply:
It is not possible to set obj to self in the arguments.

Reply:
That's shame. We can still remove this by abstracting the recursion to a separate method. But that adds one more method just for this and even now it is not part of the class API.
So I'm leaving that decision on you. It is fine for me even with this form.